packages/components/src/Button/icon.ts

```diff
@@ -64,10 +64,37 @@ export const iconMargins = (props: ButtonProps) => {
   }
 }
 
+export const iconSizes = (props: ButtonProps) => {
+  let iconSize = 16
+  switch (props.size) {
+    case 'xsmall':
+      iconSize = 12
+      break
+    case 'small':
+      iconSize = 14
+      break
+    case 'large':
+      iconSize = 18
+      break
+    default:
```

Comment:
I prefer seeing all cases handled in the switch even if there functionally equivalent – makes it easier to catch issues where the default value might get changed in one place in the code but not another.

```suggestion
    case 'medium':
    default:
```